pkgs/applications/networking/irc/weechat/default.nix

```diff
@@ -78,52 +79,83 @@ let
         platforms = stdenv.lib.platforms.unix;
       };
     };
 in if configure == null then weechat else
   let
     perlInterpreter = perl;
-    config = configure {
-      availablePlugins = let
-          simplePlugin = name: {pluginFile = "${weechat.${name}}/lib/weechat/plugins/${name}.so";};
-        in rec {
-          python = {
-            pluginFile = "${weechat.python}/lib/weechat/plugins/python.so";
-            withPackages = pkgsFun: (python // {
-              extraEnv = ''
-                export PYTHONHOME="${pythonPackages.python.withPackages pkgsFun}"
-              '';
-            });
-          };
-          perl = (simplePlugin "perl") // {
+    availablePlugins = let
+        simplePlugin = name: {pluginFile = "${weechat.${name}}/lib/weechat/plugins/${name}.so";};
+      in rec {
+        python = {
+          pluginFile = "${weechat.python}/lib/weechat/plugins/python.so";
+          withPackages = pkgsFun: (python // {
             extraEnv = ''
-              export PATH="${perlInterpreter}/bin:$PATH"
+              export PYTHONHOME="${pythonPackages.python.withPackages pkgsFun}"
             '';
-          };
-          tcl = simplePlugin "tcl";
-          ruby = simplePlugin "ruby";
-          guile = simplePlugin "guile";
-          lua = simplePlugin "lua";
+          });
         };
+        perl = (simplePlugin "perl") // {
+          extraEnv = ''
+            export PATH="${perlInterpreter}/bin:$PATH"
+          '';
+        };
+        tcl = simplePlugin "tcl";
+        ruby = simplePlugin "ruby";
+        guile = simplePlugin "guile";
+        lua = simplePlugin "lua";
       };
 
-    inherit (config) plugins;
+    config = configure { inherit availablePlugins; };
+
+    plugins = config.plugins or (builtins.attrValues availablePlugins);
 
     pluginsDir = runCommand "weechat-plugins" {} ''
       mkdir -p $out/plugins
       for plugin in ${lib.concatMapStringsSep " " (p: p.pluginFile) plugins} ; do
         ln -s $plugin $out/plugins
       done
     '';
-  in (writeScriptBin "weechat" ''
-    #!${stdenv.shell}
-    export WEECHAT_EXTRA_LIBDIR=${pluginsDir}
-    ${lib.concatMapStringsSep "\n" (p: lib.optionalString (p ? extraEnv) p.extraEnv) plugins}
-    exec ${weechat}/bin/weechat "$@"
-  '') // {
-    name = weechat.name;
-    unwrapped = weechat;
-    meta = weechat.meta;
+
+    init = let
+      init = builtins.replaceStrings [ "\n" ] [ ";" ] (config.init or "");
+
+      mkScript = drv: lib.flip map drv.scripts (script: "/script load ${drv}/share/${script}");
+
+      scripts = builtins.concatStringsSep ";" (lib.foldl (scripts: drv: scripts ++ mkScript drv)
+        [ ] (config.scripts or []));
+    in init + scripts;
+
+    mkWeechat = bin: (writeScriptBin bin ''
+      #!${stdenv.shell}
+      index=0
+      next=
+      declare -a list=("$@")
+
+      # check if $@ contains a `--run-command` option. If so,
+      # it will be merged with the `init' property from `weechat.override'.
```

Comment:
Now that this is [fixed upstream](https://github.com/weechat/weechat/issues/1248#event-1831302139), we could instead cherry-pick the upstream solution as a patch. Either that or add a comment to the `version` attribute to remind whoever updates weechat next to remove this.

Reply:
now this was fast :D 
Unfortunately the patch doesn't apply onto v2.2, I'll try to rebase it, but it seems to be a bit of work 
as several translation files are affected.

Reply:
I was definitely positively surprised, nice when upstream is so reactive :)

Yeah no reason to rebase it I think. You could make use of fetchpatch's `includes` if the translation files are the only issues (that really are a *lot* of translation files). Otherwise the comment to the version is also fine.

Reply:
ah dammit, `src/core/weechat.c` contains some conflicts as well. But I'll think about this tomorrow, I'll catch some sleep now :)

Reply:
No need to spend too much time on this then. We just need to make sure that the code doesn't stay around forever and for that the comment should be enough.